R/read-results.R

```diff
@@ -0,0 +1,38 @@
+
+#' Read test results from CSV files in a directory.
+#' @param test_output_dir path to a directory containing test output files. Each
+#'   CSV file should have an accompanying JSON file. See [input-formats].
+#' @return List of two elements, a tibble with test result ("results") and
+#'   metadata information for each test result set ("info").
+#' @importFrom glue glue
+#' @importFrom purrr map reduce
+#' @importFrom readr read_csv
+#' @importFrom stringr str_replace fixed
+#' @importFrom tibble add_column
+#' @export
+read_csv_test_results <- function(test_output_dir) {
+  csv_files <- list.files(test_output_dir, pattern = "\\.csv$", full.names = TRUE)
+  json_files <- str_replace(csv_files, "\\.csv$", ".json")
+
+  for (.j in json_files) {
+    if (!fs::file_exists(.j)) {
+      rlang::abort(
+        glue("Expected JSON info file does not exist: {.j}"),
+        "mrgvalidate_missing_result_info")
+    }
+  }
+
+  results <- map(
+    csv_files,
+    ~{
+      read_csv(.x, show_col_types = FALSE) %>%
```

Comment:
This was failing for me and I realized it's because `show_col_types` was just added in [2.0.0 which came out last week](https://github.com/tidyverse/readr/releases/tag/v2.0.0). Let's get rid of this so we don't have to constrain to `readr >= 2.0.0`. You can use `col_types = readr::cols()` to accomplish the same thing (albeit slightly uglier).

As a sidenote: this is why we have bbr (and some other packages) set up to test again an "oldest supported" and newest possible set of dependencies in CI. That would be great to get set up for here too, but probably not high on the priority list given our deadlines.

Reply:
Good catch.  Thanks.